### ISSUE TITLE
Consider `&mdash` and `-` as the same character

### ISSUE THREAD

**Issue by savchenko** [keepassxreboot/keepassxc#5735]
## Summary
Consider `&mdash` and `-` as the same character when performing auto-type.

## Examples
Nothing to show here.

## Context
In the recent Firefox release, @mozilla have made a change of replacing regular dash with emdash:
- _Before:_ "Login - Mozilla Firefox"
- _After:_ "Login &mdash; Mozilla Firefox"

As the result, saved auto-type patterns are now invalid.

**Comment by keepassxc-bugtracking202011** [comment]
I have the same issue https://github.com/keepassxreboot/keepassxc/issues/5736
based on the changes implemented by mozilla https://bugzilla.mozilla.org/show_bug.cgi?id=1279647.

Really annoying change by mozilla impacting my hundreds of users with a lot of stored auto-type accounts.

**Comment by savchenko** [comment]
Reading Bugzilla I wonder if we may ask @mkaply why the change was necessary in the first place.

**Comment by droidmonkey** [comment]
So this is in the window title?

**Comment by savchenko** [comment]
> So this is in the window title?

Yes: 
![image](https://user-images.githubusercontent.com/300146/99956808-88d93e80-2d7e-11eb-8477-152ea40bd278.png)

**Comment by droidmonkey** [comment]
Your best bet at this time is to change your window association so they replace the emdash with *

**Comment by savchenko** [comment]
> change your window association 

Manually change record by record or there is an interface to apply this en masse?

**Comment by droidmonkey** [comment]
You'd have to go to each record

**Comment by mkaply** [comment]
This was done for aesthetic purposes. The emdash just looks better.

**Comment by h-h-h-h** [comment]
> This was done for aesthetic purposes. The emdash just looks better.

It's also practical. Just recently I had it to do with quite a number of plaintext tabs where I quickly needed to read the filenames. The titles showed variable-length paths, filename and browser name. With the short hyphen, this was more annoying, because it doesn't visually separate as well as a longer dash and thus claims more brain power to do.

**Comment by mkaply** [comment]
Yeah, lots of pages use a - in their title. This makes it clear which is the Firefox bit and which isn't.

**Comment by hifi** [comment]
There's a silver lining for power users. An upcoming feature for 2.7.0 will make auto-type entries globally searchable from the selection window which can possibly make window associations unnecessary.

**Comment by r3times** [comment]
> This was done for aesthetic purposes. The emdash just looks better.

"Thanks".

**Comment by WM4362** [comment]
> Your best bet at this time is to change your window association so they replace the emdash with *

I'm surprised this comment from @droidmonkey hasn't received more love... this is THE fix so far as I'm concerned.

I previously had separate Window Associations for each browser. Now I only have one association ending with '*' and it works for ANY browser!

That's a great feature I wish I'd known about yonks ago, thanks for pointing it out @droidmonkey!
